src/target/source/codegen_c_host.cc

```diff
@@ -93,18 +94,46 @@ void CodeGenCHost::AddFunction(const PrimFunc& f) {
   }
 }
 
-void CodeGenCHost::PrintFuncPrefix() {  // NOLINT(*)
-  stream << "#ifdef __cplusplus\n"
-         << "extern \"C\"\n"
-         << "#endif\n"
-         << "TVM_DLL int32_t";
+void CodeGenCHost::GenerateForwardFunctionDeclarations(String global_symbol,
+                                                       const Array<PrimExpr>& args) {
+  for (auto& func_already_defined : GetFunctionNames()) {
+    if (global_symbol == func_already_defined) {
+      return;
+    }
+  }
+  this->PrintFuncPrefix(fwd_decl_stream);
+  fwd_decl_stream << " " << global_symbol << "(";
+  for (size_t i = 1; i < args.size(); ++i) {
```

Comment:
Would it make sense to have a `PrintFunctionPrototype`? 

It seems there's more logic here: https://github.com/apache/tvm/blob/a70c0649a895cf600e600bb3a1f49f6269aa71db/src/target/source/codegen_c.cc#L92-L119 which could apply?

Reply:
Wouldn't it be better to share the same logic though? Then we can guarantee the forward declarations will always match the function prints later? Otherwise we'd have to update in both places should we change how we print the function definition.

Reply:
Resolved offline, this requires more information to be threaded and is out of scope here, can refactor later if necessary